### ISSUE TITLE
NumericUpDown does not update when editing text directly

### ISSUE THREAD

**Issue by thargy** [NaBian/HandyControl#1283]
### Describe the bug

When editing the number in the numeric display, rather than using the spinners, the source is never updated.  Pressing a spinner will act on the pre-edited version of the number.

### Steps to reproduce the bug

Add a NumericUpDown, e.g.

```xaml
<hc:NumericUpDown hc:InfoElement.Placeholder="Please enter days worked per year."
                  hc:InfoElement.Title="Days worked"
                  hc:InfoElement.TitlePlacement="Left"
                  Value="{Binding Path=Settings.Days, Mode=TwoWay, UpdateSourceTrigger=PropertyChanged}"
                  Minimum="1" Maximum="365" DecimalPlaces="1" Increment="0.5" ValueFormat="F1"
                  Style="{StaticResource NumericUpDownExtend}" />
```

### Expected behavior

Changing the value directly (rather than using spinners) should update the bound value.

### NuGet package version

HandyControl 3.4.0

### IDE

Visual Studio 2022

### Framework type

.Net 6.0

### Windows version

Windows 11 (22000)

### Additional context

This issue was also reported [here](https://github.com/ghost1372/HandyControls/issues/165)

**Comment by NaBian** [comment]
We haven't released 3.4.0 yet, it will be fixed in 3.4.0.

**Comment by thargy** [comment]
@NaBian Thanks, but your fix will potentially introduce a new bug:
https://github.com/HandyOrg/HandyControl/blob/c513f5c6f5f2c5a56a403eb1e038c90aa9a7f427/src/Shared/HandyControl_Shared/Controls/Input/NumericUpDown.cs#L48-L52

Should contain:
```csharp
_textBox.LostFocus -= TextBox_LostFocus;
```

Raised #1287 to track.
